pkg/controller/installation/core_controller.go

```diff
@@ -882,6 +882,10 @@ func isOpenshiftOnAws(install *operator.Installation, ctx context.Context, clien
 }
 
 func updateInstallationForOpenshiftNetwork(i *operator.Installation, o *configv1.Network) error {
+	// If CNI plugin is specified and not Calico then skip any CalicoNetwork initialization
+	if i.Spec.CNI != nil && i.Spec.CNI.Type != operator.PluginCalico {
```

Comment:
I don't think this is really correct - we sometimes want a CalicoNetworkSpec even if not using Calico CNI.

Reply:
If you recall, this was one of my concerns about this particular expression of the API - `CalicoNetwork` != `cni.type:Calico`

Reply:
With only the CNI plugin being able to be specified right now I don't think there are any cases where we'd want CalicoNetwork. Did we allow CalicoNetwork to be specified with the EKS, AKS, or GKE providers?